### DISCUSSION TITLE
How to monitor the gas consumtion direct or indirect from a vaíllant EcoTecPlus (bai)

### DISCUSSION THREAD

**Discussion by ostauss** [john30/ebusd#725] [Q&A]
Dear experts, 
Ii already control my EcoTecPlus using the ebusd to set values on my f37 thermostat. This works “fine”. 
Next, I want to use the ebusd to monitor the heating and the water boiler. 
My biggest Problem here is, I did not “understand” the values in deep – so I don't know, which values I need 
 to monitor and visualize...
At least, I would like to have an indicator, when the System has a high or low consumption.  
Best would be to read the gas consumption in m³ ;-) 

Maybe we can start a discussion about useful values and the technical meaning on these values. 
Thanks in advance

**Comment by JeeCee1** [comment]
If you have `ebusctl` running you should be able to read energy usage for ```PrFuelSumHc``` for Hc (heating) and ```PrFuelSumHwc``` for Hwc (Hot Tapwater) in kWh. Depending on the source of your gas, 1m3 roughly equals 10 kWh.

Note that these values not be red directly from the heater. In my case I read them from the controller/thermostat (VRC700). I don't know if they are available on a F37, they might be n/a or differently named.

Also, the controller/thermostat needs to be wired to the ebus, not mounted directly in the front housing of the heater (it uses a different bus than ebus)

**Comment by Erbit** [comment]
I have EcoTecPlus. I read data from:
PrEnergySumHc1 - (heating)
PrEnergySumHwc1 - (hot water)

Both meters do not give values in m3 or in kW.

It was experimentally established:
Hc1=747.351 = 1m3
Hwc1 = 672.730 = 1m3

I don't know why there are such converters and in what units Vaillant counts, but it works.

**Reply by Moneybox76** [reply]
Do you know if these values to calculate are the same for all Vaillant systems?

**Reply by pepsonEL** [reply]
I have EcoTecPlus VSC206 4-5 and by one day Hc1 grow up about 36 and for me your solution is no correct.

**Reply by Dinth** [reply]
Ive been comparing PrEnergySumHc1/Hcw1 data with kWh data from my smart meter and it doesn't seem that PrEnergySum* are linear

**Reply by Erbit** [reply]
Sorry, I have EcoTecPRO, not EcoTecPlus. Model VCWPL 226/5-3.

For me, it's linear.  I don't know if these calculated values are the same for all Vaillant systems but it seems to work on several Vaillant boilers. I refer to the source of information -> https://blog-techniczny.pl/2018/04/08/pomiar-zuzycia-gazu-przez-raspberry-pi-i-ebus/

**Reply by pepsonEL** [reply]
You are from PL ?

**Reply by JeeCee1** [reply]
> I have EcoTecPlus VSC206 4-5 and by one day Hc1 grow up about 36 and for me your solution is no correct.

That would be 3.6 m3. Depending on your outside temperature and heating habits that could be correct, although I have 206 kWh today (300m2 heated).

**Reply by JeeCee1** [reply]
> Ive been comparing PrEnergySumHc1/Hcw1 data with kWh data from my smart meter and it doesn't seem that PrEnergySum* are linear

The sum of those plus a little heater losses should be linear, if not I don't know what the cause is. Do you cook on gas?

**Reply by Dinth** [reply]
So, firstly i checked the formula posted by someone either here or HA forums and it was completely different than mine. I blamed the different caloric value of my gas and started measuring it myself - on days when i was not cooking on gas. I was not summing Hc1 and Hcw1 as it was summer and heating was off - just took Hcw1 into consideration. On each day i have been making a separate spreadsheet with measurements during the day (after updates of data from the smart meter, as it is sent every 30 minutes instead of live) and found out that during the day the ratio of SumHcw1 and used gas from my meter is growing. Either ive got some serious leak (not according to the gas engineer who was testing my gas pressure a couple of years ago) or the data from Vaillant really is not linear.

**Reply by Erbit** [reply]
> You are from PL ?

No, from Poland ;)

**Reply by pepsonEL** [reply]
I am also from Poland. Did you have FB or are you on group Home Assistant Polska on FB ?

**Comment by pepsonEL** [comment]
But how we can create sensor in HA to calculate it for m3 ?

**Reply by JeeCee1** [reply]
That's more a question for the [HA forum](https://community.home-assistant.io/), but first have a look at the documentation for a [template sensor](https://www.home-assistant.io/integrations/template/).

My code:

```
template:
    #================
    #=== Conversion m3 to kWh
    #=== z-Zahl * Brennwert
    #================
  - sensor:
    - unique_id: gas_consumption_kwh
      name: Gas Consumption kWh
      state: >
        {{ states('input_number.gas_consumption')|float(0) * 10 }}
      icon: mdi:gas-burner
      unit_of_measurement: kWh
```

**Reply by pepsonEL** [reply]
but my Vaillant in multimatic controller show in kWh not in m3.

**Reply by Dinth** [reply]
If you want to calculate volume of gas from kWh, i believe that you need to know a caloric value of your gas first. In my country suppliers write this on each months bill.
